examples/settings/settings_main.c

```diff
@@ -0,0 +1,276 @@
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sys/mount.h>
+
+#include "system/settings.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * settings_main
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  int                  ret;
+  int                  fd;
+  enum settings_type_e stype;
+  char                 path[50];
+  char                 *teststr = "";
+  int                  testval = 0x5aa5;
+  char                 readstr[CONFIG_SETTINGS_VALUE_SIZE];
+  enum storage_type_e  storage_type = STORAGE_BINARY;
+  bool                 flag_present = false;
+  struct stat          sbuf;
+
+  ASSERT(0);
+  if ((argc < 1) || *argv[1] == 0 || *(argv[1] + 1) == 0)
+    {
+      goto print_help;
+    }
+
+  if (*argv[1] == '-')
+    {
+      flag_present = true;
+      if (*(argv[1] + 1) == 'b')
+        {
+          storage_type = STORAGE_BINARY;
+        }
+      else if (*(argv[1] + 1) == 't')
+        {
+          storage_type = STORAGE_TEXT;
+        }
+      else
+        {
+          goto print_help;
+        }
+    }
+
+  if (flag_present && argc < 2)
```

Comment:
I think that this is an impossible condition.

Reply:
It checks that there aren't too many arguments. If the app is called as "settings -b -rubbish" it traps for that; even though it may be unnecessary. But it is strict and therefore OK I think?